### ISSUE TITLE
Many fewer things are needed when SCREAM_LIB_ONLY=True

### ISSUE THREAD

**Issue by jgfouca** [E3SM-Project/scream#547]
The cxx->f90 bindings and probably most or all of the *_f90.[ch]pp files.

**Comment by bartgol** [comment]
Can we piggy back on this PR, and change `SCREAM_LIB_ONLY` into `SCREAM_ENABLE_TESTS`. The latter is a common cmake var (`${PROJECT_NAME}_ENABLE_TESTS`), that people would likely understand, and maybe even try to grep for when they want to figure out how to turn on/off the tests.

Obviously, all the conditionals would have to be flipped (i.e., SCREAM_ENABLE_TESTS = NOT SCREAM_LIB_ONLY).

**Comment by PeterCaldwell** [comment]
@jgfouca - did this ever get fixed?

**Comment by bartgol** [comment]
I don't think so. And it would be nice to have it. Especially on GPU, where `p3_functions_f90.cpp` alone takes 4-5 minutes to build..

**Comment by jgfouca** [comment]
@PeterCaldwell , it did not. I will look at it.

**Comment by PeterCaldwell** [comment]
When do we want SCREAM_LIB_ONLY=True? Is this when we're building for doing a run rather than for unit testing? If so, I like Luca's idea of calling this flag SCREAM_ENABLE_TESTS=False. And is this a capability we need before we can run a v1 compset?

**Comment by bartgol** [comment]
By default, SCREAM_LIB_ONLY is TRUE if it's a CIME build, and FALSE otherwise. But the used can override this.

Enabling this option (or, equivalently, disabling a replacement var like SCREAM_ENABLE_TESTS) is not strictly speaking necessary for v1 runs. We can build tests and e3sm.exe at the same time. It is, however, a bit more expensive in terms of compilation time (and I'm not just talking about executables: think about all the p3 f90-cxx bridges, which are only needed for testing). That said, I would like to have tests built in CIME nightly/AT testing, cause they might expose some configuration bug that only shows with that option (but nevertheless lurks underneath for the lib only scenario).
